packages/x-date-pickers/src/internals/hooks/useDesktopPicker/useDesktopPicker.tsx

```diff
@@ -149,7 +151,8 @@ export const useDesktopPicker = <
         />
         <PickersPopper
           role="dialog"
-          anchorEl={inputRef.current}
+          placement="bottom-start"
+          anchorEl={fieldRef.current}
```

Comment:
I agree, the popper should hide the helper text. We fixed the exact same bug with the autocomplete in the first few months.

Reply:
As long as there is no `startAdornment` indeed it is horizontally-equivalent

But if the user sets the icon on the `startAdornment`, then setting the ref on the field has an advantage: 

![image](https://user-images.githubusercontent.com/3309670/211739745-7af8baaa-17b3-4a06-9502-e4c764b229fb.png)

We can have the best of both world by putting the ref to the `InputProps.ref` 

![image](https://user-images.githubusercontent.com/3309670/211740011-d8c0338e-c83c-4396-9a79-96ce3500a8b4.png)


Do you see any major downside here ?

Reply:
> Do you see any major downside here ?

Hmm... Nothing major. Just the fact that it would be a breaking change for anyone who is using custom `Input` slot. 🤔 
But I guess it is a minority of users in any case and as long as we provide information in migration doc—it should be fine. 🤞 🙈

Reply:
>  it would be a breaking change for anyone who is using custom Input slot. 

Not if we merge the refs from `componentsProps.input.InputProps.ref` and our internal ref before passing it to the field.

Reply:
>  it would be a breaking change for anyone who is using custom Input slot. 

The migration from `renderInput` to `components.input` and `componentsProps.input` will not be codemodabe for advanced scenario, so people will have to check it by hand anyway right ?

We can merge  `componentsProps.input.InputProps.ref` and our internal ref before passing it to the field.
For refs defined in custom `components.input`, people will indeed have to be careful but like for any other prop one could say (if they define a `onClick` and we pass one as well, they should make sure to call our).

Reply:
> The migration from `renderInput` to `components.input` and `componentsProps.input` will not be codemodabe for advanced scenario, so people will have to check it by hand anyway right ?

Yes, fair point. I was not expecting to have it codemodable.
I just wanted to stress the fact that we already have A LOT of breaking changes and quite a bit of those will not be codemodable. Was just trying to understand if we really need this change on top of all it. 😉

Reply:
Right

So if we summarize, it's a breaking change if people were passing fully custom input like so:

```tsx
const CustomInput = props => {
  const InputRef = React.useRef()

  return (
    <TextField {...props} InputProps={{ ...props.InputProps, ref: InputRef }} />
  )
}


<DatePicker renderInput={props => <CustomInput {...props} />} />
```

Because they will replace it (automatically via codemod or not) with:

```tsx
const CustomInput = props => {
  const InputRef = React.useRef()

  return (
    <TextField {...props} components={{ textField: CustomInput }} />
  )
}


<DatePicker renderInput={props => <CustomInput {...props} />} />
```

____

If they were passing the ref from the parent:

```tsx
const InputRef = React.useRef()

<DatePicker 
  renderInput={props => (
    <TextField {...props} InputProps={{ ...props.InputProps, ref: InputRef }} />
  ) 
/>
```

Then we will advise them to migrate to 

```tsx
const CustomInput = props => {
  const InputRef = React.useRef()

  return (
    <TextField {...props} componentsProps={{ textField: { InputProps: { ref: InputRef  }}}} />
  )
}


<DatePicker renderInput={props => <CustomInput {...props} />} />
```

Which should work fine

____

That raises an interesting question about what a breaking change is.
If tomorrow we start passing a new prop to the input, is it a breaking change ?

For example, we did not consider #5651 to be a breaking change.
Yet, if people were passing a custom `onMouseUp`, starting to pass one ourselves broke their app, unless they did

```
<TextField 
  onMouseUp={e => {
    props.onMouseUp?.(e);
    handleMouseUp(e);
  }} 
/>
```

Which I think they should always do.

Reply:
Fair points. 👌 
Regarding the fact of what we treat as a breaking change and what isn't—that can probably always be improved, because you sometimes might not even imagine what kind of configuration combination users might have. 🤔 

Regarding which case I though this change would break is the usage of [native `input`](https://next.mui.com/x/react-date-pickers/date-picker/#custom-input-component) or any other non MUI component usage. Or am I missing something? 🤔

Reply:
> Regarding which case I though this change would break is the usage of [native input](https://next.mui.com/x/react-date-pickers/date-picker/#custom-input-component) or any other non MUI component usage. Or am I missing something? thinking

Indeed
We should probably add a warning in the future headless doc on how to manage the ref.

I'll wait for @oliviertassinari opinion on this one.
Clearly this change is not super important.

Reply:
I think that it should use the Input's root to position the popup, like in https://github.com/mui/material-ui/blob/777445676712cadbea0775aa26a5dfc69442c760/packages/mui-material/src/Autocomplete/Autocomplete.js#L556-L557.

Reply:
I modified it to use the `InputProps.ref` instead :+1: